minorminer/utils/parallel_embeddings.py

```diff
@@ -0,0 +1,630 @@
+        pos = dnx.pegasus_layout(G)
+        dnx.draw_pegasus(**draw_kwargs)
+    elif topology == "zephyr":
+        pos = dnx.zephyr_layout(G)
+        dnx.draw_zephyr(**draw_kwargs)
+    else:
+        pos = nx.spring_layout(G)
+        nx.draw_networkx(**draw_kwargs)
+    if len(edge_color_dict) > 0:
+        # Recolor specific edges on top of the original graph
+        nx.draw_networkx_edges(
+            G,
+            pos=pos,
+            edgelist=list(edge_color_dict.keys()),
+            edge_color=list(edge_color_dict.values()),
+            width=1,
+            edge_cmap=cmap,
+            ax=ax,
+        )
+    return node_color_dict, edge_color_dict
+
+
+def shuffle_graph(
+    G: nx.Graph, seed: Union[int, np.random.RandomState, np.random.Generator] = None
+) -> nx.Graph:
+    """Shuffle the node and edge ordering of a networkx graph.
+
+    For embedding methods that operate as a function of the node or edge
+    ordering this can force diversification in the returned embeddings. Note
+    that special orderings that encode graph structure or geometry may enhance
+    embedder performance (shuffling may lead to longer search times).
+
+    Args:
+        G: A networkx graph
+        seed: When provided, is used to shuffle the order of nodes and edges in
+        the source and target graph. This can allow sampling from otherwise deterministic routines.
+    Returns:
+        nx.Graph: The same graph with modified node and edge ordering.
+    """
+    prng = np.random.default_rng(seed)
+    nodes = list(G.nodes())
+    prng.shuffle(nodes)
+    edges = list(G.edges())
+    prng.shuffle(edges)
+    _G = nx.Graph()
+    _G.add_nodes_from(nodes)
+    _G.add_edges_from(edges)
+    return _G
+
+
+def embeddings_to_array(embs: list, node_order=None, as_ndarray=False):
+    """Convert list of embeddings into an array of embedding values
+
+    Args:
+        embs: A list of embeddings, each list entry in the
+            form of a dictionary with integer values.
+        node_order: An iterable giving the ordering of
+            variables in each row. When not provided variables are ordered to
+            match the first embedding :code:`embs[0].keys()`. node_order
+            can define any subset of the source graph nodes (embedding
+            keys).
+        as_ndarray: The list is cast to an ndarray, this requires that
+            the embedding is 1 to 1, that the embedding values (nodes of
+            a target graph) can be cast as numpy numeric types.
+
+    Raises:
+        ValueError: If `embs` is empty and `node_order` not provided.
+
+    Returns:
+        An embedding matrix; each row defines an embedding ordered
+            by node_order.
+    """
+    if node_order is None:
+        if len(embs) == 0:
+            if as_ndarray:
+                raise ValueError("shape of ndarray cannot be inferred")
+            else:
+                return []
+        else:
+            node_order = embs[0].keys()
```

Comment:
IMO looks cleaner without the elses here.
```suggestion
            return []

        node_order = embs[0].keys()
```